internal/sdkv2provider/resource_cloudflare_zone_settings_override_migrate.go

```diff
@@ -0,0 +1,59 @@
+package sdkv2provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceCloudflareZoneSettingsOverrideV0() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"settings": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: resourceCloudflareZoneSettingsSchemaV0,
+				},
+			},
+
+			"initial_settings": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: resourceCloudflareZoneSettingsSchemaV0,
+				},
+			},
+		},
+	}
+}
+
+func resourceCloudflareZoneSettingsOverrideStateUpgradeV1(
+	_ context.Context,
+	rawState map[string]interface{},
+	_ interface{},
+) (map[string]interface{}, error) {
+	if rawState == nil {
+		return nil, fmt.Errorf("could not upgrade cloudflare_zone_settings_override from v0 to v1: state is nil")
+	}
+
+	settings, ok := rawState["settings"].(map[string]interface{})
+	if settings == nil || !ok {
+		return nil, fmt.Errorf("could not upgrade cloudflare_zone_settings_override from v0 to v1: state does not have a `settings` field")
+	}
+
+	initialSettings, ok := rawState["initial_settings"].(map[string]interface{})
+	if settings == nil || !ok {
```

Comment:
the conditional here is incorrect

```suggestion
	if initialSettings == nil || !ok {
```

Reply:
addressed in bb0417383b678572a15f272648fbe8cff4fa987f with other changes